artman/conductors/cloudtasks_conductor.py

```diff
@@ -0,0 +1,147 @@
+
+from artman.cli import main
+from artman.utils.logger import logger
+
+
+def run(queue_name):
+    task_client = _create_tasks_client()
+    if True:  # TODO change to when
+        pull_task_response = _pull_task(task_client, queue_name)
+        tasks = pull_task_response.get('tasks', [])
+        if not tasks:
+            # Sleep for 30 seconds if there is no tasks returned.
+            logger.info('There is no pending task. Sleep for 30 seconds.')
+            time.sleep(30)
+        for task in tasks:
+            tmp_reporoot, artman_user_config = _prepare_dir()
+            try:
+                logger.info('Starting to execute task %s' % task)
+                _execute_task(artman_user_config, task)
+                _ack_task(task_client, task)
+                logger.info('Task execution finished for %s' % tasks)
+            except:
+                logger.info('Task execution failed for %s' % tasks)
+                _cancel_task_lease(task_client, task)
+            finally:
+                logger.info('Cleanup tmp directory %s' % tmp_reporoot)
+                _cleanup_tmp_dir(tmp_reporoot)
+
+
+def _create_tasks_client():
+    credentials = GoogleCredentials.get_application_default()
+    with open(
+        os.path.join(os.path.dirname(__file__), 'cloudtasks.json'), 'r') as f:
+            return build_from_document(f.read(), credentials=credentials)
+
+
+def _pull_task(task_client, queue_name):
+    body = {
+      "maxTasks": 1,
+      "leaseDuration": {"seconds": 300, "nanos": 0},  # Expire after 300 secs.
+      "responseView": "FULL",
+      "name": "%s" % queue_name
+    }
+    tasks = task_client.projects().locations().queues().tasks().pull(
+        name=queue_name, body=body).execute()
+    logger.info('Pulling tasks request returned %s' % tasks)
+    return tasks
+
+
+def _ack_task(task_client, task):
+    body = {'scheduleTime': task['scheduleTime']}
+    response = task_client.projects().locations().queues().tasks().acknowledge(
+        name=task['name'],
+        body=body).execute()
+    logger.info('Acknowledge task request returned %s' % response)
+    return response
+
+
+def _cancel_task_lease(task_client, task):
+    body = {'scheduleTime': task['scheduleTime'], "responseView": "FULL"}
+    response = task_client.projects().locations().queues().tasks().cancelLease(
+        name=task['name'],
+        body=body).execute()
+    logger.info('Cancel task request returned %s' % response)
+    return response
+
+
+def _execute_task(artman_user_config, task):
+    """Execute the remote artman tasks.
+
+    It execute the artman command with a customized artman user config and
+    additional pipeline arguments."""
+    task_payload = base64.b64decode(task['pullTaskTarget']['payload'])
+    artman_args = task_payload.decode("utf-8").split(' ')
+    artman_args.append('--user-config')
+    artman_args.append(artman_user_config)
+    # TODO(ethanbao): No longer need to override toolkit_path once the
```

Comment:
This _should_ be done already, but I think there might be a bug.

If `${REPOROOT}/toolkit` is still appearing in config, it needs to become `${TOOLKIT}`.